File: web-app/src/lib/features/features/history.ts
import type { Feature } from '../types';

export const historyFeatures = {
  'history-get-detail': {
    id: 'history-get-detail',
    title: 'Get Claude History Detail',
    description: 'Retrieves the full detail of a single Claude conversation history entry by ID.',
    rpcIds: ['history:get-detail'],
    componentPaths: [],
    testIds: ['TestGetHistoryDetail_EmptyID'],
    status: 'stable',
    since: '1.0.0',
  },
  'history-get-messages': {
    id: 'history-get-messages',
    title: 'Get Claude History Messages',
    description: 'Retrieves the individual messages from a Claude conversation history entry.',
    rpcIds: ['history:get-messages'],
    componentPaths: [],
    testIds: ['TestGetHistoryMessages_EmptyID'],
    status: 'stable',
    since: '1.0.0',
  },
  'history-list': {
    id: 'history-list',
    title: 'List Claude History',
    description: 'Lists all available Claude conversation history entries.',
    rpcIds: ['history:list'],
    componentPaths: [],
    testIds: ['TestListClaudeHistory_EmptyDir'],
    status: 'stable',
    since: '1.0.0',
  },
  'history-search': {
    id: 'history-search',
    title: 'Search Claude History',
    description: 'Searches Claude conversation history entries by query string.',
    rpcIds: ['history:search'],
    componentPaths: [],
    testIds: ['TestSearchHistory_EmptyQuery'],
    status: 'stable',
    since: '1.0.0',
  },
} as const satisfies Record<string, Feature>;

File: server/features/history.go
package features

import "github.com/tstapler/stapler-squad/server/featureregistry"

var HistoryGetDetail = featureregistry.Feature{
	ID:          "history-get-detail",
	Title:       "Get Claude History Detail",
	Description: "Retrieves the full detail of a single Claude conversation history entry by ID.",
	RPCIDs:      []string{"history:get-detail"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

var HistoryGetMessages = featureregistry.Feature{
	ID:          "history-get-messages",
	Title:       "Get Claude History Messages",
	Description: "Retrieves the individual messages from a Claude conversation history entry.",
	RPCIDs:      []string{"history:get-messages"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

var HistoryList = featureregistry.Feature{
	ID:          "history-list",
	Title:       "List Claude History",
	Description: "Lists all available Claude conversation history entries.",
	RPCIDs:      []string{"history:list"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

var HistorySearch = featureregistry.Feature{
	ID:          "history-search",
	Title:       "Search Claude History",
	Description: "Searches Claude conversation history entries by query string.",
	RPCIDs:      []string{"history:search"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

func init() {
	featureregistry.Register(HistoryGetDetail)
	featureregistry.Register(HistoryGetMessages)
	featureregistry.Register(HistoryList)
	featureregistry.Register(HistorySearch)
}
